src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -921,4 +922,52 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise
+     */
+    public void validateDates() throws BadRequestException {
+        // All fields are optional, so there is no error if none of them is set
+        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null) {
+            return;
+        }
+        boolean validDates = true;
+        if (getReleaseDate() == null) {
+            if (getDueDate() == null) {
+                // There is an assesmentDueDate, but no dueDate / releaseDate
+                validDates = false;
+            }
+            else {
+                // There is an dueDate/ assesmentDueDate, but no releaseDate
+                if (getAssessmentDueDate() != null) {
+                    validDates = getAssessmentDueDate().isAfter(getDueDate());
+                }
+            }
+            if (!validDates) {
+                throw new BadRequestException();
+            }
+        }
+        else {
+            if (getDueDate() == null) {
+                // There is an releaseDate / assesmentDueDate, but no dueDate
+                if (getAssessmentDueDate() != null) {
+                    validDates = false;
+                }
+            }
+            else {
+                // We have all dates set
+                if (getAssessmentDueDate() != null) {
+
+                    validDates = getReleaseDate().isBefore(getDueDate()) && getDueDate().isBefore(getAssessmentDueDate());
+                }
+                // There is an releaseDate / dueDate, but no assesmentDueDate
+                else {
+                    validDates = getReleaseDate().isBefore(getDueDate());
+                }
+            }
+            if (!validDates) {
+                throw new BadRequestException();
+            }
+        }
```

Comment:
I feel like this whole thing could be refactored, so there are less nested if / else statements.

Reply:
refactored it with 2 small helper methods